tests/unit/mesh/test_ncmesh.cpp

```diff
@@ -297,6 +297,228 @@ TEST_CASE("pNCMesh PA diagonal",  "[Parallel], [NCMesh]")
+   // solutions match exactly.
+   auto check_l2_projection = [&exact_soln](ParMesh& pmesh, Mesh& smesh, int order)
+   {
+      MPI_Barrier(MPI_COMM_WORLD); // Get our ducks in a row.
+
+      REQUIRE(pmesh.GetGlobalNE() == smesh.GetNE());
+      REQUIRE(pmesh.Dimension() == smesh.Dimension());
+      REQUIRE(pmesh.SpaceDimension() == smesh.SpaceDimension());
+
+      // Make an H1 space, then a mass matrix operator and invert it.
+      // If all non-conformal constraints have been conveyed correctly, the
+      // resulting DOF should match exactly on the serial and the parallel
+      // solution.
+
+      H1_FECollection fec(order, smesh.Dimension());
+      ConstantCoefficient one(1.0);
+      FunctionCoefficient rhs_coef(exact_soln);
+
+      constexpr double linear_tol = 1e-16;
+
+      // serial solve
+      auto serror = [&]
+      {
+         FiniteElementSpace fes(&smesh, &fec);
+         // solution vectors
+         GridFunction x(&fes);
+         x = 0.0;
+
+         LinearForm b(&fes);
+         b.AddDomainIntegrator(new DomainLFIntegrator(rhs_coef));
+         b.Assemble();
+
+         BilinearForm a(&fes);
+         a.AddDomainIntegrator(new MassIntegrator(one));
+         a.Assemble();
+
+         SparseMatrix A;
+         Vector B, X;
+
+         Array<int> empty_tdof_list;
+         a.FormLinearSystem(empty_tdof_list, x, b, A, X, B);
+
+#ifndef MFEM_USE_SUITESPARSE
+         // 9. Define a simple symmetric Gauss-Seidel preconditioner and use it to
+         //    solve the system AX=B with PCG.
+         GSSmoother M(A);
+         PCG(A, M, B, X, -1, 500, linear_tol, 0.0);
+#else
+         // 9. If MFEM was compiled with SuiteSparse, use UMFPACK to solve the system.
+         UMFPackSolver umf_solver;
+         umf_solver.Control[UMFPACK_ORDERING] = UMFPACK_ORDERING_METIS;
+         umf_solver.SetOperator(A);
+         umf_solver.Mult(B, X);
+#endif
+
+         a.RecoverFEMSolution(X, b, x);
+         return x.ComputeL2Error(rhs_coef);
+      }();
+
+      auto perror = [&]
+      {
+         // parallel solve
+         ParFiniteElementSpace fes(&pmesh, &fec);
+         ParLinearForm b(&fes);
+
+         ParGridFunction x(&fes);
+         x = 0.0;
+
+         b.AddDomainIntegrator(new DomainLFIntegrator(rhs_coef));
+         b.Assemble();
+
+         ParBilinearForm a(&fes);
+         a.AddDomainIntegrator(new MassIntegrator(one));
+         a.Assemble();
+
+         HypreParMatrix A;
+         Vector B, X;
+         Array<int> empty_tdof_list;
+         a.FormLinearSystem(empty_tdof_list, x, b, A, X, B);
+
+         HypreBoomerAMG amg(A);
+         HyprePCG pcg(A);
+         amg.SetPrintLevel(-1);
+         pcg.SetTol(linear_tol);
+         pcg.SetMaxIter(500);
+         pcg.SetPrintLevel(-1);
+         pcg.SetPreconditioner(amg);
+         pcg.Mult(B, X);
+         a.RecoverFEMSolution(X, b, x);
+         return x.ComputeL2Error(rhs_coef);
+      }();
+
+      constexpr double test_tol = 1e-9;
```

Comment:
The values of `serror` and `perror` are on the order of 1e-4, so I wonder why their difference is on the order of 1e-10. Maybe it is due to the PCG solver used in my build without suitesparse. Did anyone try with suitesparse and get more digits of accuracy? This doesn't have to be answered for the PR review, just curious.